src/SparkConduit.sol

```diff
@@ -167,22 +159,20 @@ contract SparkConduit is UpgradeableProxied, ISparkConduit, IInterestRateDataSou
     }
 
     function requestFunds(bytes32 ilk, address asset, uint256 amount) external ilkAuth(ilk) {
-        uint256 liquidityAvailable = IERC20(asset).balanceOf(IPool(pool).getReserveData(asset).aTokenAddress);
-        require(liquidityAvailable == 0, "SparkConduit/non-zero-liquidity");
-
-        // Convert asset amount to shares
-        uint256 newRequestedShares = amount.rayDiv(IPool(pool).getReserveNormalizedIncome(asset));
+        // TODO: Update this to avoid DoS vector
```

Comment:
As we discussed before I think we can just add another function that will do a withdraw + request funds all in one atomic call.

Reply:
Yeah fair I'll leave TODO and we'll add in another PR